src/Tags/Taggable.hs

```diff
@@ -130,554 +125,86 @@ subtractLocation a b = subtractRange (locationByteRange a) (locationByteRange b)
 
 -- Instances
 
-instance ( Apply Show1 fs, Apply Functor fs, Apply Foldable fs, Apply Traversable fs, Apply Taggable fs) => Taggable (Sum fs) where
-  docsLiteral a = apply @Taggable (docsLiteral a)
-  snippet x = apply @Taggable (snippet x)
-  symbolName = apply @Taggable symbolName
-
-instance (Taggable a) => Taggable (TermF a Location) where
-  docsLiteral l t = docsLiteral l (termFOut t)
-  snippet ann t = snippet ann (termFOut t)
-  symbolName t = symbolName (termFOut t)
-
-instance Taggable Syntax.Context where
-  snippet ann (Syntax.Context _ (Term (In subj _))) = Just (subtractLocation ann subj)
-
-instance Taggable Declaration.Function where
-  docsLiteral Python (Declaration.Function _ _ _ (Term (In _ bodyF)))
+instance (TaggableBy strategy t, strategy ~ TaggableInstance t) => Taggable t where
+  docsLiteral = docsLiteral' @strategy
+  snippet = snippet' @strategy
+  symbolName = symbolName' @strategy
```

Comment:
I opted to use `TypeApplications` & `AllowAmbiguousTypes` instead of passing a proxy around. I feel pretty good about this.